packages/vx-axis/src/axis/AnimatedTicksRenderer/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { animated, useTransition, interpolate } from 'react-spring';
+import cx from 'classnames';
+import { Text } from '@vx/text';
+
+import Orientation from '../../constants/orientation';
+import { TicksRendererProps, AxisScale } from '../../types';
+import useTickTransitionConfig from './useTickTransitionConfig';
+
+export default function TicksRenderer<Scale extends AxisScale>({
+  hideTicks,
+  horizontal,
+  orientation,
+  scale,
+  tickClassName,
+  tickLabelProps: allTickLabelProps,
+  tickStroke = '#222',
+  tickTransform,
+  ticks,
+}: TicksRendererProps<Scale>) {
+  const transitionConfig = useTickTransitionConfig({ horizontal, scale });
+  const animatedTicks = useTransition(ticks, tick => `${tick.value}-${horizontal}`, {
+    unique: true,
+    ...transitionConfig,
+  });
+
+  return animatedTicks.map(({ item, key, props }, index) => {
+    // @ts-ignore react-spring types don't handle fromX, etc.
+    const { fromX, toX, fromY, toY, opacity } = props;
+    const tickLabelProps = allTickLabelProps[index] ?? allTickLabelProps[0] ?? {};
```

Comment:
I couldn't come up with a great solution for this. when tick values update, there may not be a `tickLabelProps` for that index.

ultimately the weirdness comes from the `Axis` component trying to find the max tick label font size to correctly offset the axis label (will comment on this). maybe we should deprecate that logic and clean this up?